homeassistant/components/mobile_app/__init__.py

```diff
@@ -1,355 +1,63 @@
-            _LOGGER.error("Error updating mobile_app registration: %s", ex)
-            return Response(status=200)
-
-        return json_response(safe_device(data))
-
-
-def supports_encryption():
-    """Test if we support encryption."""
-    try:
-        import nacl   # noqa pylint: disable=unused-import
-        return True
-    except OSError:
-        return False
-
-
-def safe_device(device: dict):
-    """Return a device without webhook_id or secret."""
-    return {
-        ATTR_APP_DATA: device[ATTR_APP_DATA],
-        ATTR_APP_ID: device[ATTR_APP_ID],
-        ATTR_APP_NAME: device[ATTR_APP_NAME],
-        ATTR_APP_VERSION: device[ATTR_APP_VERSION],
-        ATTR_DEVICE_NAME: device[ATTR_DEVICE_NAME],
-        ATTR_MANUFACTURER: device[ATTR_MANUFACTURER],
-        ATTR_MODEL: device[ATTR_MODEL],
-        ATTR_OS_VERSION: device[ATTR_OS_VERSION],
-        ATTR_SUPPORTS_ENCRYPTION: device[ATTR_SUPPORTS_ENCRYPTION],
-    }
-
-
-def register_device_webhook(hass: HomeAssistantType, store, device):
-    """Register the webhook for a device."""
-    device_name = 'Mobile App: {}'.format(device[ATTR_DEVICE_NAME])
-    webhook_id = device[CONF_WEBHOOK_ID]
-    webhook.async_register(hass, DOMAIN, device_name, webhook_id,
-                           partial(handle_webhook, store))
+REQUIREMENTS = ['PyNaCl==1.3.0']
 
 
-async def async_setup(hass, config):
+async def async_setup(hass: HomeAssistantType, config: ConfigType):
     """Set up the mobile app component."""
-    conf = config.get(DOMAIN)
-
     store = hass.helpers.storage.Store(STORAGE_VERSION, STORAGE_KEY)
     app_config = await store.async_load()
     if app_config is None:
-        app_config = {}
-
-    hass.data[DOMAIN] = app_config
+        app_config = {DATA_DELETED_IDS: [], DATA_REGISTRATIONS: {}}
 
-    for device in app_config.values():
-        register_device_webhook(hass, store, device)
+    if hass.data.get(DOMAIN) is None:
+        hass.data[DOMAIN] = {DATA_DELETED_IDS: [], DATA_REGISTRATIONS: {}}
 
-    if conf is not None:
-        hass.async_create_task(hass.config_entries.flow.async_init(
-            DOMAIN, context={'source': config_entries.SOURCE_IMPORT}))
+    hass.data[DOMAIN][DATA_DELETED_IDS] = app_config[DATA_DELETED_IDS]
+    hass.data[DOMAIN][DATA_REGISTRATIONS] = app_config[DATA_REGISTRATIONS]
+    hass.data[DOMAIN][DATA_STORE] = store
 
-    hass.http.register_view(DevicesView(store))
-
-    return True
+    for registration in app_config[DATA_REGISTRATIONS].values():
+        setup_registration(hass, store, registration)
 
+    register_http_handlers(hass, store)
+    register_websocket_handlers(hass)
+    register_deleted_webhooks(hass, store)
 
-async def async_setup_entry(hass, entry):
-    """Set up an mobile_app entry."""
     return True
 
 
-class DevicesView(HomeAssistantView):
-    """A view that accepts device registration requests."""
+@bind_hass
+def async_registrations(hass: HomeAssistantType, component: str) -> list:
```

Comment:
This is not being used. Let's drop it. It might be in the way of us migrating to config entries.